### PR TITLE
Jasmin - tokenizer

### PR DESCRIPTION
**Pull request by jasmingoh**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [{
+        "type": "node",
+        "request": "launch",
+        "name": "Launch Program",
+        "program": "${workspaceFolder}\\src\\parser\\Tokenizer.js"
```
Comment (jasmingoh) [inline .vscode/launch.json]: this helps to run the tokenize method quickly on the debug tab

File: src/parser/Tokenizer.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        this.tokens = []
+        this.currentTokenIdx = 0;
+
+        this.tokenize();
+    }
+
+    tokenize() {
+        const lines = this.program.split("\n") || [];
+        lines.forEach(line => {
+            // if ':' exists in line
```
Comment (danielanatolie) [inline src/parser/Tokenizer.js]: No need for this comment.

File: src/parser/Tokenizer.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    checkNext() {
+        if (this.currentTokenIdx < this.tokens.length) {
+            return this.tokens[this.currentTokenIdx];
+        } else {
+            return false;
+        }
+    }
+
+    getNext() {
+        // this "consumes" the token
```
Comment (danielanatolie) [inline src/parser/Tokenizer.js]: No need for this comment.

**Comment by jasmingoh** [comment]
@danielanatolie #14

**Pull request merged**